fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/service/SavingsSchedularServiceImpl.java

```diff
@@ -65,23 +86,61 @@ public void postInterestForAccounts() throws JobExecutionException {
         Integer initialSize = 500;
         Integer totalPageSize = 0;
         List<Throwable> errors = new ArrayList<>();
-        do {
-            PageRequest pageRequest = PageRequest.of(page, initialSize);
-            Page<SavingsAccount> savingsAccounts = this.savingsAccountRepository.findByStatus(ACTIVE.getValue(), pageRequest);
-            for (SavingsAccount savingsAccount : savingsAccounts.getContent()) {
-                try {
-                    this.savingAccountAssembler.assignSavingAccountHelpers(savingsAccount);
-                    boolean postInterestAsOn = false;
-                    LocalDate transactionDate = null;
-                    this.savingsAccountWritePlatformService.postInterest(savingsAccount, postInterestAsOn, transactionDate);
-                } catch (Exception e) {
-                    LOG.error("Failed to post interest for Savings with id {}", savingsAccount.getId(), e);
-                    errors.add(e);
-                }
+
+        if (this.globalConfigurationRepository.findOneByName("allow-split-interest-posting-on-savings").isEnabled()) {
+            final ScheduledJobDetail scheduledJobDetail = this.schedularWritePlatformService.retrieveJobByNameAndNodeId(this.nodeId,
+                    JobName.POST_INTEREST_FOR_SAVINGS.toString());
+            final String officeId = this.jobParameterRepository
```

Comment:
you dont need to fetch jobparams directly from here. job params are suposed to be injected from jobregister service. Ref: https://github.com/apache/fineract/pull/1971/files#diff-00a8c0bc5ea386690c4ae4abc7d5a8820b2230c01bff72d24cb1c695656ccf21